app/src/main/java/com/m/services/dataSync/TaskActivity.java

```diff
@@ -48,6 +52,15 @@ public void addTask(Item task) {
                 itemList.add(task);
                 updateUi();
             }
+
+            @Override
+            public void onFailure(String errorMessage) {
+                if (errorMessage.equals("HTTP 403 Forbidden")) {
```

Comment:
is it possible to get the status code from the exception, rather than relying on the message? The problem with messages is they could change a lot, and when that happens this will break the logic here. Status code is normally more stable.